### ISSUE TITLE
TupleHash expectedResults.json sample files include the customization string for MCT when it shouldn't

### ISSUE THREAD

**Issue by blackbird1999** [usnistgov/ACVP-Server#211]
**environment**
Demo

**testSessionId**
N/A

**vsId**
0 (see https://github.com/usnistgov/ACVP-Server/blob/master/gen-val/json-files/TUPLEHASH-128-1.0/expectedResults.json)

**Algorithm registration**
N/A

**Endpoint in which the error is experienced**
N/A

**Expected behavior**
It should match the ACVP specification and not include the customization string as documented in https://pages.nist.gov/ACVP/draft-celi-acvp-xof.html

**Additional context**
https://pages.nist.gov/ACVP/draft-celi-acvp-xof.html section 8.3 details what the response data should include. It does not list the customization string as being included. It should just be the md and outLen. The ACVP server is outputting the customization string as well. It's unclear what the server expects from other clients. Maybe it's properly ignoring them? Maybe it requires them?

**Comment by blackbird1999** [comment]
Also incorrectly outputs the intermediate tuples.

**Comment by markowitz-isc** [comment]
I agree that per the spec, _resultsArray_ should not have _tuple_ and _customization_ properties, but in case anyone is interested, I want to point out that those values are _Output[i]_ and _Customization_ respectively, at the start of each inner loop, while _md_ is, of course, the value of _OutputJ[j]_ after the inner loop. 

The main problem here, though, is that the line:

    workingBits = Left(T[i-1][0] || ZeroBits(288), 288);

in the TupleHash MCT pseudocode (see [draft-celi-acvp-xof.html#name-tuplehash-monte-carlo-test](https://pages.nist.gov/ACVP/draft-celi-acvp-xof.html#name-tuplehash-monte-carlo-test)) should be

    workingBits = Left(Output[i-1] || ZeroBits(288), 288);

with Output[0] properly initialized to the input tuple (rather putting that in T[0][0])!

As the pseudocode stands, there is no feedback between the output of the hash function and _workingBits_ -- in fact, its first 9 bytes and those of _T[i][0]_ remain invariant, so _tupleSize_ is also invariant. Clearly that is not what the server is executing.

**Comment by livebe01** [comment]
@blackbird1999 thanks for mentioning these issues with https://github.com/usnistgov/ACVP-Server/blob/master/gen-val/json-files/TUPLEHASH-128-1.0/expectedResults.json. For the longest time, https://github.com/usnistgov/ACVP-Server/blob/master/gen-val/json-files/ were just example files that we referenced internally for our own purposes. Some of the capabilities registrations and expected results files, for example, happened to contain extra information/stuff that's not _supposed_ to be there, but wouldn't necessarily cause an error on the server. But now that these are public domain and folks are looking at and referencing them, we should clean these up/make them correct. I'll fix TupleHash MCT and verify the other XOFs right now and we can see about doing a more comprehensive review of all the sample files in the future. 

Also, I noticed that we don't have any example test results for TupleHash and ParallelHash [here](https://pages.nist.gov/ACVP/draft-celi-acvp-xof.html#name-example-test-results-json-o). I'll check into getting some examples into that doc as well.
